src/eduid_action/mfa/idp.py

```diff
@@ -101,8 +100,17 @@ def check_authn_result(idp_app, user, ticket, actions):
     for this in actions:
         if isinstance(this.result, dict):
             idp_app.logger.debug('Action {} authn result: {}'.format(this, this.result))
-            if this.result.get('success', False):
-                idp_app.logger.debug('Removing successful MFA action')
-                idp_app.actions_db.remove_action_by_id(this.action_id)
+            if this.result.get('success', False) == True:
```

Comment:
Tror att jämförelse med is är bättre här eftersom du förväntar dig ett True och inget annat.

Reply:
In [1]: "True" is True
Out[1]: False

In [2]: "True" == True
Out[2]: False

In [3]:

men jag håller med om att det kan göras snyggare - fixar